### DISCUSSION TITLE
Setting custom username?

### DISCUSSION THREAD

**Discussion by cheulyop** [coder/code-server#2051] [Q&A] [answered]
Maybe this is a simple mistake, but I'm having some trouble setting a customer username, as it seems that we can do so since v3.5.0:
> You can now set DOCKER_USER when invoicing the docker container to set a custom username #2001

However, when I try to set a custom username by passing DOCKER_USER as an environment variable when running a docker container for the first time, it gives me an error `usermod: user 'coder' does not exist`. Here is the command I used to set up the container:
```console
$ docker run --gpus all -it -e DOCKER_USER=cheulyop -p 6010:6006 -p 8084:8080 -u "$(id -u):$(id -g)" -v ~/projects:/home/coder/projects -v /data:/home/coder/data -v ~/.config:/home/coder/.config --ipc=host cheulyop/deepo-code-server:latest
```
What am I doing wrong here?

**Comment by nhooyr** [comment, marked as answer]
Hi @cheulyop 

You're not running our docker image. Ours is `codercom/code-server`, you're using `cheulyop/deepo-code-server`.

**Reply by nhooyr** [reply]
Ah nvm, your image must be based on ours.

**Reply by nhooyr** [reply]
Oh, it could be your image already changes the username?

**Reply by cheulyop** [reply]
Hi @nhooyr. Yes, the image I'm using is based on yours, particularly on [ci/release-image/Dockerfile](https://github.com/cdr/code-server/blob/v3.5.0/ci/release-image/Dockerfile). My Dockerfile is the same as yours line-by-line, except the base image which I use my own.

Given that, I'm guessing these lines may be what's causing the problem?
https://github.com/cdr/code-server/blob/b509063e143bbf74b74ec295260c4fd5f6332f71/ci/release-image/Dockerfile#L27-L28

or maybe this:
https://github.com/cdr/code-server/blob/b509063e143bbf74b74ec295260c4fd5f6332f71/ci/release-image/Dockerfile#L42

**Reply by nhooyr** [reply]
I think I found the bug, see https://github.com/cdr/code-server/issues/2056 and https://github.com/cdr/code-server/pull/2057

**Reply by nhooyr** [reply]
Thanks for reporting!
